### PR TITLE
Add snapshot deletion to cluster cleanup

### PR DESCRIPTION
**Pull request by sd109**

### CODE REVIEW

File: capi_janitor/openstack/operator.py [hunk 1]
```diff
@@ -232,20 +243,36 @@ async def on_openstackcluster_event(type, name, namespace, meta, spec, **kwargs)
...
         check_volumes = False
         volumes_annotation_value = meta.get("annotations", {}).get(
             VOLUMES_ANNOTATION,
             VOLUMES_ANNOTATION_DEFAULT
         )
         if volumes_annotation_value == VOLUMES_ANNOTATION_DELETE:
             check_volumes = True
+            async for snapshot in snapshots_for_cluster(snapshots_detail, name):
+                if snapshot.status != "deleting":
+                    await snapshots.delete(snapshot.id)
```
Comment (mkjpryor) [inline capi_janitor/openstack/operator.py:261]: Do we also need the 400 protection here for the chained case that you mentioned?

File: capi_janitor/openstack/operator.py [hunk 2]
```diff
@@ -256,6 +283,8 @@ async def on_openstackcluster_event(type, name, namespace, meta, spec, **kwargs)
             raise ResourcesStillPresentError("loadbalancers", name)
         if check_volumes and not await empty(volumes_for_cluster(volumes_detail, name)):
             raise ResourcesStillPresentError("volumes", name)
+        if check_volumes and not await empty(snapshots_for_cluster(snapshots_detail, name)):
+            raise ResourcesStillPresentError("snapshots", name)
```
Comment (mkjpryor) [inline capi_janitor/openstack/operator.py:287]: Can we use a separate `check_snapshots` here please? This avoids rechecking the snapshots in the case where backup is not enabled.

**Review by mkjpryor — CHANGES_REQUESTED** [review]
Looks good. Just a couple of minor things.

File: capi_janitor/openstack/operator.py (hunk 1 above)
Comment (sd109) [reply capi_janitor/openstack/operator.py:261]: Yes, probably safer to add it.

**Review by mkjpryor — APPROVED** [review]
LGTM

**Pull request merged**